src/main/java/sparrow/storage/Storage.java

```diff
@@ -0,0 +1,173 @@
+package sparrow.storage;
+
+import sparrow.data.task.Task;
+import sparrow.data.task.Todo;
+import sparrow.data.task.Deadline;
+import sparrow.data.task.Event;
+import sparrow.data.task.TaskList;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the file used to store the task list.
+ */
+public class Storage {
+
+    private static final String DEFAULT_FILE_PATH = "Sparrow.txt";
+
+    private final Path path;
+
+    public Storage() throws Exception {
+        this(DEFAULT_FILE_PATH);
+    }
+
+    public Storage(String filePath) throws Exception {
+        path = Path.of(filePath);
+        if (!isValidPath(path)) {
+            throw new Exception("Invalid file extension");
+        }
+
+    }
+
+    private boolean isValidPath(Path filePath) {
+        return filePath.toString().endsWith(".txt");
+    }
+
+    /**
+     * Returns list of tasks loaded from hard disk.
+     * If no file found, returns an empty list.
+     * @return Task list.
+     */
+    public TaskList loadFromFile() {
+        File f = new File(path.toString());
+
+        if (f.exists()) {
+            try {
+                return decodeTaskList(Files.readAllLines(path));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            try {
+                f.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        return new TaskList();
+    }
+
+    public void saveToFile(TaskList tasks) {
+        try {
+            Files.write(path, encodeTaskList(tasks));
+        } catch (IOException e) {
+            // TODO create exception for this
+            System.out.println("Error saving to file" + e.toString());
+        }
+    }
+
+    public TaskList decodeTaskList(List<String> encodedTaskList) {
+        ArrayList<Task> tasks = new ArrayList<>();
+        for (String encodedTask : encodedTaskList) {
+            if (!encodedTask.isBlank()) {
+                tasks.add(stringToTask(encodedTask));
+            }
+        }
+        return new TaskList(tasks);
+    }
+
+    public List<String> encodeTaskList(TaskList tasks) {
+        List<String> encodedTaskList = new ArrayList<>();
+        for (Task task : tasks.getTasks()) {
+            encodedTaskList.add(taskToString(task));
+        }
+        return encodedTaskList;
+    }
+
+    /**
+     * Converts user input into a Task.
+     * @param input User input to be converted.
```

Comment:
I think you can add a line in between the parameters and the description of the method